### PR TITLE
feat(gce): UI for zone selection.

### PR DESCRIPTION
**Pull request by jtk54**
Depends on https://github.com/spinnaker/clouddriver/pull/2201.

### CODE REVIEW

**Comment by jtk54** [comment]
image

**Comment by jtk54** [comment]
image

**Comment by jtk54** [comment]
image

**Comment by danielpeach** [comment]
Instead of the `Select explicit zones` checkbox, couldn't you just always have that `ui-select` available, but prepopulate it?

Is there a limit to the number of zones you can select?

**Comment by jtk54** [comment]
We want to preserve the 'just use the default' option, which is why there are two views with a checkbox.

The limit is just the number of zones we get from Clouddriver for that region.

File: app/scripts/modules/google/src/serverGroup/configure/wizard/zones/zoneSelector.directive.html [hunk 1]
```diff
@@ -22,4 +22,14 @@
       </li>
     </ul>
   </div>
+  <div class="col-md-7" ng-if="vm.command.selectZones">
+    <ui-select multiple
```
Comment (danielpeach) [inline app/scripts/modules/google/src/serverGroup/configure/wizard/zones/zoneSelector.directive.html]: Should have a minimum of 1 zone selected I think - from the Clouddriver PR, if you select 0 zones, I think you'll end up with the default 3.

File: app/scripts/modules/google/src/serverGroup/configure/wizard/zones/regionalSelector.directive.html [hunk 2]
```diff
@@ -4,5 +4,8 @@
     <label><input type="checkbox" ng-model="vm.command.regional"/>
       Distribute instances across multiple zones
     </label>
+    <label ng-if="vm.command.regional"><input type="checkbox" ng-model="vm.command.selectZones"/>
+      Select explicit zones
```
Comment (danielpeach) [inline app/scripts/modules/google/src/serverGroup/configure/wizard/zones/regionalSelector.directive.html]: `Select zones explicitly`, maybe? `explicit zones` sounds a little different...

File: app/scripts/modules/google/src/serverGroup/configure/wizard/zones/zoneSelector.directive.html (hunk 1 above)
Comment (jtk54) [reply app/scripts/modules/google/src/serverGroup/configure/wizard/zones/zoneSelector.directive.html]: Agreed, added validation on this.

**Pull request merged**